bundles/org.openhab.binding.myuplink/src/main/java/org/openhab/binding/myuplink/internal/command/AbstractWriteCommand.java

```diff
@@ -0,0 +1,151 @@
+ */
+package org.openhab.binding.myuplink.internal.command;
+
+import static org.openhab.binding.myuplink.internal.MyUplinkBindingConstants.EMPTY;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.measure.MetricPrefix;
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.api.Request;
+import org.openhab.binding.myuplink.internal.Utils;
+import org.openhab.binding.myuplink.internal.handler.MyUplinkThingHandler;
+import org.openhab.binding.myuplink.internal.model.ValidationException;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * base class for all write commands. common logic should be implemented here
+ *
+ * @author Alexander Friese - initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractWriteCommand extends AbstractCommand {
+    private final Logger logger = LoggerFactory.getLogger(AbstractWriteCommand.class);
+
+    protected final Channel channel;
+    protected final Command command;
+
+    /**
+     * the constructor
+     */
+    public AbstractWriteCommand(MyUplinkThingHandler handler, Channel channel, Command command,
+            RetryOnFailure retryOnFailure, ProcessFailureResponse processFailureResponse,
+            JsonResultProcessor resultProcessor) {
+        super(handler, retryOnFailure, processFailureResponse, resultProcessor);
+        this.channel = channel;
+        this.command = command;
+    }
+
+    /**
+     * helper method for write commands that extracts value from command.
+     *
+     * @return value as String without unit.
+     */
+    protected String getCommandValue() {
+        if (command instanceof QuantityType<?> quantityCommand) {
+            // this is necessary because we must not send the unit to the backend
+            Unit<?> unit = quantityCommand.getUnit();
+            QuantityType<?> convertedType;
+            if (unit.isCompatible(SIUnits.CELSIUS)) {
+                convertedType = quantityCommand.toUnit(SIUnits.CELSIUS);
+            } else if (unit.isCompatible(Units.KILOWATT_HOUR)) {
+                convertedType = quantityCommand.toUnit(Units.KILOWATT_HOUR);
+            } else if (unit.isCompatible(Units.LITRE_PER_MINUTE)) {
+                convertedType = quantityCommand.toUnit(Units.LITRE_PER_MINUTE);
+            } else if (unit.isCompatible(tech.units.indriya.unit.Units.WATT)) {
+                convertedType = quantityCommand.toUnit(MetricPrefix.KILO(tech.units.indriya.unit.Units.WATT));
+            } else {
+                logger.info("automatic conversion of unit '{}' to myUplink expected unit not supported.",
```

Comment:
Looking at the code these loggers seem to be the last issues. Please check your logging levels. 
info => hardly used it only when really usefull for the user started componend or listening on interface etc. Itlooks like most of the times you use info, warn should be used.
sometimes you log to debug but also add the payload/json. You can use debug for the url /endpoint metadata etc, but message content (like json) should go to trace. 
For reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging

https://www.openhab.org/docs/developer/guidelines.html#f-logging
For reference

Reply:
fixed it